stock_release_channel_warehouse_calendar/models/stock_release_channel.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2025 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from datetime import timedelta
+
+import pytz
+
+from odoo import models
+
+from odoo.addons.stock_release_channel import delivery_date_generator
+
+
+class StockReleaseChannel(models.Model):
+    _inherit = "stock.release.channel"
+
+    @delivery_date_generator(step="preparation")
+    def _next_delivery_date_warehouse_calendar(self, delivery_date, partner=None):
+        """Get the next valid delivery date respecting warehouse calendar
+
+        The preparation date must be during warehouse working hours given by
+        the calendar on the warehouse.
+        """
+        calendar = self.warehouse_id.calendar_id
+        if not calendar:
+            while True:
+                delivery_date = yield delivery_date
+        wh_tz = pytz.timezone(self.warehouse_id.partner_id.tz or "UTC")
+        batch_delta = timedelta(days=61)
+        while True:
+            delivery_date = delivery_date.astimezone(pytz.utc)
+            work_intervals = calendar._work_intervals_batch(
+                delivery_date, delivery_date + batch_delta, tz=wh_tz
+            )[False]
+            for begin_dt_tz, end_dt_tz, _attendance in work_intervals:
+                while delivery_date <= end_dt_tz:
+                    if delivery_date < begin_dt_tz:
+                        delivery_date = begin_dt_tz
+                    delivery_date = yield delivery_date.astimezone(pytz.utc).replace(
+                        tzinfo=None
+                    )
+                    delivery_date = delivery_date.astimezone(pytz.utc)
```

Comment:
I'm sorry, but I don't get why this method would expect input from `generator.send()`.
Also, I feel like wh.calendar_id is meant to compute expedition date, not delivery date.